Code/Mantid/Framework/MDAlgorithms/src/IntegrateMDHistoWorkspace.cpp

```diff
@@ -106,7 +106,7 @@ Mantid::coord_t getPrecisionCorrectedCoordinate(Mantid::coord_t position) {
   const auto deviation = fabs(nearest - position)/nearest;
   const auto tolerance = 1e-6;
   Mantid::coord_t coordinate(position);
-  if (deviation < tolerance) {
+  if (fabs(deviation) < tolerance) {
```

Comment:
The fabs should be at the at the definition of deviation. It should be a matter of additional braces.